plugins/kubernetes/app/controllers/kubernetes_roles_controller.rb

```diff
@@ -1,74 +1,48 @@
 class KubernetesRolesController < ApplicationController
-  before_action :authorize_deployer!
-  before_action :project
-  before_action :find_role, only: [:show, :edit, :update]
+  include ProjectLevelAuthorization
 
-  def new
-    @kubernetes_role = Kubernetes::Role.new(project: project, ram: 512, cpu: 0.2, replicas: 1)
-  end
-
-  def create
-    @kubernetes_role = @project.roles.build(new_role_params)
-    @kubernetes_role.deploy_strategy = 'rolling_update'    # temporarily hardcoded
-    @kubernetes_role.save
-
-    respond_to do |format|
-      format.html do
-        if @kubernetes_role.persisted?
-          redirect_to project_kubernetes_roles_path(@project)
-        else
-          render :new, status: 422
-        end
-      end
-
-      format.json do
-        render json: {}, status: @kubernetes_role.persisted? ? 200 : 422
-      end
-    end
-  end
+  before_action :authorize_project_deployer!
```

Comment:
You should probably wrap the new/create/update methods under the project_admin privs.